src/components/RewardWrapper/index.tsx

```diff
@@ -214,15 +221,27 @@ export function RewardWrapper({
 
     let address: `0x${string}` | undefined
 
-    if (account.address) {
+    /**
+     * handles https://github.com/HyperPlay-Gaming/product-management/issues/801
+     * Sometimes wagmi does not establish a connection but useAccount returns the address.
+     * We need to check that the switch chain method exists before proceeding with claiming.
+     */
+    let connectionHasSwitchChain = false
+    if (config.state.current) {
+      const currentConnection = config.state.connections.get(
+        config.state.current
+      )
+      connectionHasSwitchChain = !!currentConnection?.connector.switchChain
+    }
+
+    if (account.address && connectionHasSwitchChain) {
       address = account.address
```

Comment:
looking at their code, it looks like switchChain is not supported by all the connectors https://github.com/wevm/wagmi/blob/main/packages/core/src/connectors/createConnector.ts#L64

if that's the case, then we'd never enter this if clause